2024/YoctoProjectDeveloperDay/practical-sec-tee.md

```diff
@@ -0,0 +1,994 @@
+│   ├── optee
+│   │   ├── optee-client
+│   │   │   ├── tee-supplicant@.service
+│   │   │   └── tee-supplicant.sh
+│   │   ├── optee-client_4.1.0.bb
+│   │   ├── optee-client.inc
+│   │   ├── optee-examples_4.1.0.bb
+│   │   ├── optee.inc
+(...)
+│   │   ├── optee-os_4.1.0.bb
+│   │   ├── optee-os.inc
+(...)
+```
+]
+.right-code-column50[
+```bash
+meta-arm-bsp
+(...)
+├── recipes-bsp
+(...)
+│   ├── trusted-firmware-a
+(...)
+│   │   ├── fiptool-native_2.8.6.bb
+│   │   ├── tf-a-tests_2.8.0.bb
+│   │   ├── tf-a-tests_%.bbappend
+│   │   ├── trusted-firmware-a_%.bbappend
+(...)
+│   ├── trusted-firmware-m
+(...)
+│   │   ├── trusted-firmware-m_%.bbappend
+│   │   └── trusted-firmware-m-corstone1000.inc
+(...)
+├── recipes-security
+│   ├── optee
+(...)
+│   │   ├── optee-client
+│   │   │   ├── tee-supplicant@.service
+│   │   │   └── tee-supplicant.sh
+(...)
+│   │   ├── optee-ftpm_%.bbappend
+│   │   ├── optee-os_4.%.bbappend
+(...)
+```
+]
+
+???
+
+- Time for this slide: TODO
+- Idea/goal of this slide: To show that meta-arm has everithing neccessary and
+  even more.
+- What to say:
+    - TODO
+- Notes:
+    - Porting problems:
+        - TODO
+
+---
+
+# Using vendor binaries
+
+.center[ <img src="./img/vendor-binaries-meme.png" height="350px"> ]
+
+.center[ <span style="font-size:30px;">There's some kind of mess here!</span>]
+
+???
+
+- Time for this slide: TODO
+- Idea/goal of this slide: To say that though vendor supplied binaries were
+  sopposed to make development easier, they made it even worse.
+- What to say:
+    - TODO
+- Notes:
+    - Weak points:
+        - You have to trust vendor binaries;
+        - You have to use vendor-specific tools, that are often not suitable for
+          your environment, are very outdated, and lack support;
+        - You do not have source code, so, debugging becames nearly impossible;
+        - If you have a small contract or do not have direct contact with
+          vendor, you will have to wait for support or, if you do not have time,
+          you will have to go for an endless web-digging.
+
+---
+
+# U-Boot integration
+
+1. Turn on `CONFIG_TEE` and `CONFIG_OPTEE` (and related configs, if needed)
+  configs.
+2. Link ATF (`BL31`):
+
+    ```bb
+    EXTRA_OEMAKE:append = " BL31=${DEPLOY_DIR_IMAGE}/atf.elf"
+    ```
+
+3. Link OP-TEE OS:
+
+    ```bb
+    EXTRA_OEMAKE:append = " TEE=${DEPLOY_DIR_IMAGE}/optee/tee.elf"
+    ```
+
+4. Check `binman` configuration, in case you are using other file formats (e.g.
+ raw binary).
+
+???
+
+- Time for this slide: TODO
+- Idea/goal of this slide: A list of steps on how to integrate ATF and OP-TEE
+  into bootloader.
+- What to say:
+    - TODO
+- Notes:
+    - Present some U-Boot configs;
+    - Present linking ATF and OP-TEE binaries;
+    - Present linkinng binaries vs ELF using Binman.
+    - Present `binman`-related problems.
+
+---
+
+# Linux integration
+
+Turn on `CONFIG_TEE` and `CONFIG_OPTEE` (and related, if needed) configs, and
```

Comment:
@DaniilKl, the previous and next slides have numbered lists - shouldn't we split this sentence into two points?

Reply:
I have thought about it, but if I split this sentence into a list - I will not be able to place code on the same slide, and splitting this sentence and code on two slides will, IMHO, be more inconvenient than the sentence. I am going to compensate it with speech.

Reply:
We could also shrink the code (font size by using the correct class), but I'll leave it to you to determine what is better. Closing